src/components/MyPage/Edit/Content/index.tsx

```diff
@@ -57,12 +73,44 @@ const Content = ({ profile }: Props) => {
     setProfileImage(url);
   };
 
+  const handleCheckNickname = () => {
+    const originalNickname = profile.member.nickname;
+    const changedNickname = profileState.nickname || '';
+
+    if (originalNickname === changedNickname) {
+      setIsNicknameChecked(false);
+      setIsUsableNickname(true);
+    } else {
+      setIsNicknameChecked(true);
+
+      if (changedNickname.length < 2) {
+        setIsUsableNickname(false);
+        setWarningText('닉네임은 2~15자만 가능합니다.');
+      } else {
+        nicknameCheck(changedNickname).then((response) => {
+          const isDuplicated = response.data.isDuplicated;
+
+          setIsUsableNickname(isDuplicated ? false : true);
+          setWarningText('이미 사용중인 닉네임입니다.');
+        });
+      }
+    }
+  };
```

Comment:
아래 코드처럼 조건에 걸리는 경우에 코드블럭 실행문 다음에 `return`을 넣어서 해당 조건에 대한 동작이 끝나면 함수를 빠져나가게 합니다. 그렇게 하면 중첩 조건들을 없애고  남은 경우에 대해서 동작하게 할 수 있습니다!

```suggestion
  const handleCheckNickname = () => {
    const originalNickname = profile.member.nickname;
    const changedNickname = profileState.nickname || '';

    if (originalNickname === changedNickname) {
      setIsNicknameChecked(false);
      setIsUsableNickname(true);
      return;
    }

    setIsNicknameChecked(true);

    if (changedNickname.length < 2) {
      setIsUsableNickname(false);
      setWarningText('닉네임은 2~15자만 가능합니다.');
      return;
    }

    nicknameCheck(changedNickname).then((response) => {
        const isDuplicated = response.data.isDuplicated;
        setIsUsableNickname(!isDuplicated);
        setWarningText(isDuplicated ? '이미 사용중인 닉네임입니다.' : '');
      })
      .catch((error) => {
        console.error('닉네임 체크 중 오류가 발생했습니다:', error);
        setIsUsableNickname(false);
        setWarningText('닉네임 체크 중 오류가 발생했습니다.');
      });
    };
```